dashboard/app/models/script_level.rb

```diff
@@ -623,8 +623,17 @@ def seeding_key(seed_context, use_existing_level_keys = true)
     my_lesson = seed_context.lessons.select {|l| l.id == stage_id}.first
     raise "No Lesson found for #{self.class}: #{my_key}, Lesson ID: #{stage_id}" unless my_lesson
     lesson_seeding_key = my_lesson.seeding_key(seed_context)
-
     my_key.merge!(lesson_seeding_key) {|key, _, _| raise "Duplicate key when generating seeding_key: #{key}"}
+
+    # Activity Section must be optional for now, so that we can still compute
+    # the seed key for legacy scripts. Currently, this is necessary because we
+    # output .script_json files for all legacy scripts, even though those aren't
+    # going to be used for seeding yet.
+    my_activity_section = seed_context.activity_sections.select {|s| s.id == activity_section_id}.first
+
+    # HACK: use script_level prefix so that it gets filtered out later.
```

Comment:
I think I'm just missing it but what do you mean by "script_level prefix"?

Reply:
oops -- I had called this `'script_level .activity_section.key'` at first, which was a hack to take advantage of this hack: https://github.com/code-dot-org/code-dot-org/blob/9705dd7988b0a4025212089f7385f7aecc6fdb5d/dashboard/lib/services/script_seed.rb#L183-L185 I fixed the hack and squashed the commits, but forgot to remove this comment.

Reply:
also fixed in this PR. the merge conflicts weren't too bad after all.